bindings/c/test/mako/mako.cpp

```diff
@@ -1766,28 +1651,184 @@ void printReport(Arguments const& args,
+		fmt::printf("TPS Interval:      %8d\n", args.tpsinterval);
+		fmt::printf("TPS Change:        ");
+		switch (args.tpschange) {
+		case TPS_SIN:
+			fmt::printf("%8s\n", "SIN");
+			break;
+		case TPS_SQUARE:
+			fmt::printf("%8s\n", "SQUARE");
+			break;
+		case TPS_PULSE:
+			fmt::printf("%8s\n", "PULSE");
+			break;
+		}
+	}
+	const auto tps_f = final_stats.getOpCount(OP_TRANSACTION) / duration_sec;
+	const auto tps_i = static_cast<uint64_t>(tps_f);
+	fmt::printf("Total Xacts:       %8lu\n", final_stats.getOpCount(OP_TRANSACTION));
+	fmt::printf("Total Conflicts:   %8lu\n", final_stats.getConflictCount());
+	fmt::printf("Total Errors:      %8lu\n", final_stats.getTotalErrorCount());
+	fmt::printf("Overall TPS:       %8lu\n\n", tps_i);
+
+	if (fp) {
+		fmt::fprintf(fp, "\"results\": {");
+		fmt::fprintf(fp, "\"totalDuration\": %6.3f,", duration_sec);
+		fmt::fprintf(fp, "\"totalProcesses\": %d,", args.num_processes);
+		fmt::fprintf(fp, "\"totalThreads\": %d,", args.num_threads);
+		fmt::fprintf(fp, "\"totalAsyncXacts\": %d,", args.async_xacts);
+		fmt::fprintf(fp, "\"targetTPS\": %d,", args.tpsmax);
+		fmt::fprintf(fp, "\"totalXacts\": %lu,", final_stats.getOpCount(OP_TRANSACTION));
+		fmt::fprintf(fp, "\"totalConflicts\": %lu,", final_stats.getConflictCount());
+		fmt::fprintf(fp, "\"totalErrors\": %lu,", final_stats.getTotalErrorCount());
+		fmt::fprintf(fp, "\"overallTPS\": %lu,", tps_i);
+	}
+
+	/* per-op stats */
+	printStatsHeader(args, true, true, false);
+
+	/* OPS */
+	putTitle("Total OPS");
+	if (fp) {
+		fmt::fprintf(fp, "\"totalOps\": {");
+	}
+	auto first_op = true;
+	for (auto op = 0; op < MAX_OP; op++) {
+		if ((args.txnspec.ops[op][OP_COUNT] > 0 && op != OP_TRANSACTION) || op == OP_COMMIT) {
+			putField(final_stats.getOpCount(op));
+			if (fp) {
+				if (first_op) {
+					first_op = 0;
+				} else {
+					fmt::fprintf(fp, ",");
+				}
+				fmt::fprintf(fp, "\"%s\": %lu", getOpName(op), final_stats.getOpCount(op));
+			}
+		}
+	}
+
+	/* TPS */
+	const auto tps = final_stats.getOpCount(OP_TRANSACTION) / duration_sec;
+	putFieldFloat(tps, 2);
+
+	/* Conflicts */
+	const auto conflicts_rate = final_stats.getConflictCount() / duration_sec;
+	putFieldFloat(conflicts_rate, 2);
+	fmt::print("\n");
+
+	if (fp) {
+		fmt::fprintf(fp, "}, \"tps\": %.2f, \"conflictsPerSec\": %.2f, \"errors\": {", tps, conflicts_rate);
+	}
+
+	/* Errors */
+	putTitle("Errors");
+	first_op = 1;
+	for (auto op = 0; op < MAX_OP; op++) {
+		if (args.txnspec.ops[op][OP_COUNT] > 0 && op != OP_TRANSACTION) {
+			putField(final_stats.getErrorCount(op));
+			if (fp) {
+				if (first_op) {
+					first_op = 0;
+				} else {
+					fmt::fprintf(fp, ",");
+				}
+				fmt::fprintf(fp, "\"%s\": %lu", getOpName(op), final_stats.getErrorCount(op));
+			}
+		}
+	}
+	if (fp) {
+		fmt::fprintf(fp, "}, \"numSamples\": {");
+	}
+	fmt::print("\n\n");
+
+	// Get the sketches stored in file and merge them together
+	std::array<DDSketchMako, MAX_OP> data_points;
```

Comment:
This can be huge right? Use vector, not array.

Reply:
I'll go ahead and change `ThreadStatistics` to use `std::vector` for holding the sketches as well. It's currently set it to `std::array`